### ISSUE TITLE
Problem: YL620 configuration fault

### ISSUE THREAD

**Issue by Arty1lv** [bdring/FluidNC#956]
### Wiki Search Terms

spindles configuration YL620

### Controller Board

ESP32 on a terminal block board.

### Machine Description

Gantry router xxyz. 2.2kw spindle with YL620-A VFD.

Currently it runs on grbl 1.1 on UNO board. Wanted to upgrade to esp for ability to control spindle speeds.

### Configuration file

```yaml
name: "ESP32 Dev Controller V4"
board: "ESP32 Dev Controller V4"

stepping:
  engine: RMT
  idle_ms: 250
  dir_delay_us: 1
  pulse_us: 2
  disable_delay_us: 0

axes:
  shared_stepper_disable_pin: gpio.13:low
  
  x:
    steps_per_mm: 800
    max_rate_mm_per_min: 2000
    acceleration_mm_per_sec2: 25
    max_travel_mm: 1000
    homing:
      cycle: 2
      mpos_mm: 10
      positive_direction: false
    
    motor0:
      limit_all_pin: gpio.17:low:pu
      stepstick:
        direction_pin: gpio.14
        step_pin: gpio.12
    motor1:
      null_motor:

  y:
    steps_per_mm: 800
    max_rate_mm_per_min: 2000
    acceleration_mm_per_sec2: 25
    max_travel_mm: 1000
    homing:
      cycle: 2
      mpos_mm: 10
      positive_direction: false

    motor0:
      limit_all_pin: gpio.4:low:pu
      stepstick:
        direction_pin: gpio.15
        step_pin: gpio.26
    motor1:
      null_motor:

  z:
    steps_per_mm: 800
    max_rate_mm_per_min: 2000
    acceleration_mm_per_sec2: 25
    max_travel_mm: 1000
    homing:
      cycle: 1
      mpos_mm: 10
      positive_direction: true

    motor0:
      limit_all_pin: gpio.16:low:pu
      stepstick:
        direction_pin: gpio.33
        step_pin: gpio.27
    motor1:
      null_motor:

spi:
  miso_pin: gpio.19
  mosi_pin: gpio.23
  sck_pin: gpio.18

sdcard:
  cs_pin: gpio.5
  card_detect_pin: NO_PIN

coolant:
  flood_pin: gpio.25
  mist_pin:  gpio.21

        
probe:
  pin: gpio.32:low:pu

YL620:
  uart:
    txd_pin: gpio.26
    rxd_pin: gpio.16
    rts_pin: gpio.4
    baud: 9600
    mode: 8N1
  modbus_id: 1
  tool_num: 0
  speed_map: 0=0% 0=25% 6000=25% 24000=100%
  off_on_alarm: false
```

### Startup Messages

```text
Resetting MCU
ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13220
ho 0 tail 12 room 4
load:0x40080400,len:3028
entry 0x400805e4

[MSG:INFO: FluidNC v3.7.1]
[MSG:INFO: Compiled with ESP32 SDK:v4.4.4]
[MSG:INFO: Local filesystem type is littlefs]
[MSG:INFO: Configuration file:test.yaml]
[MSG:ERR: Configuration error at /machine/YL620/uart: ERR: gpio.26 - Pin is already used.]
[MSG:ERR: Configuration error at /machine/YL620/uart: ERR: gpio.16 - Pin is already used.]
[MSG:ERR: Configuration error at /machine/YL620/uart: ERR: gpio.4 - Pin is already used.]
[MSG:ERR: Validation error at /YL620/uart: UART: TXD is undefined]
[MSG:ERR: Configuration is invalid]
[MSG:INFO: Using default configuration]
[MSG:INFO: Axes: using defaults]
[MSG:ERR: Configuration is invalid]
[MSG:INFO: STA SSID is not set]
[MSG:INFO: AP SSID FluidNC IP 192.168.0.1 mask 255.255.255.0 channel 1]
[MSG:INFO: AP started]
[MSG:INFO: WiFi on]
[MSG:INFO: Captive Portal Started]
[MSG:INFO: HTTP started on port 80]
[MSG:INFO: Telnet started on port 23]

Grbl 3.7 [FluidNC v3.7.1 (wifi) '$' for help]
[MSG:ERR: Configuration is invalid. Check boot messages for ERR's.]
```

### User Interface Software

webUI, candle

### What happened?

Wanted to upgrade from grbl on uno to fluidnc esp for ability to control spindle speeds.
Basically i've tried to connect only VFD to fluidnc to evaluate whether it's worth to migrate or not. But it seems like something is wrong with configuration. It tells that pins are already defined, so it throws these errors and after uses default config.

basically, config is a copy of 3axis_v4.yaml but with different spindle, code for spindle is copied from wiki page.

Hope it's minor problem and i could use that soft to control spindle
